src/tiled/objectiddialog.cpp

```diff
@@ -0,0 +1,176 @@
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "objectiddialog.h"
+#include "ui_objectiddialog.h"
+
+#include "documentmanager.h"
+#include "mapdocument.h"
+#include "map.h"
+#include "objectgroup.h"
+#include "mapobject.h"
+#include "utils.h"
+
+#include <QList>
+#include <QLineEdit>
+#include <QTableWidget>
+#include <QTableWidgetItem>
+#include <QTableWidgetSelectionRange>
+#include <QPushButton>
+#include <QHeaderView>
+
+namespace Tiled {
+namespace Internal {
+
+
+ObjectIdDialog::ObjectIdDialog(QWidget *parent)
+    : QDialog(parent)
+    , mUi(new Ui::ObjectIdDialog)
+    , mId(0)
+{
+    mUi->setupUi(this);
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+    Utils::restoreGeometry(this);
+
+    if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
+        QTableWidget *tableWidget = mUi->tableWidget;
+
+        QStringList headers = {QStringLiteral("ID"), QStringLiteral("Name"), QStringLiteral("Type")};
+        tableWidget->setHorizontalHeaderLabels(headers);
+        tableWidget->horizontalHeader()->setSectionResizeMode(1, QHeaderView::Stretch);
+        tableWidget->verticalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+
+        QList<ObjectGroup*> objectGroups = document->map()->objectGroups();
+
+        foreach(const ObjectGroup *group, objectGroups) {
+            foreach (const MapObject *object, group->objects())
+            {
+                tableWidget->insertRow(tableWidget->rowCount());
+                tableWidget->setItem(tableWidget->rowCount() - 1, 0, new QTableWidgetItem(QString::number(object->id())));
+                tableWidget->setItem(tableWidget->rowCount() - 1, 1, new QTableWidgetItem(object->name()));
+                tableWidget->setItem(tableWidget->rowCount() - 1, 2, new QTableWidgetItem(object->type()));
+            }
+        }
+    }
+
+    connect(mUi->lineEdit, &QLineEdit::textChanged, this, &ObjectIdDialog::onTextChanged);
+    connect(mUi->tableWidget, &QTableWidget::itemSelectionChanged, this, &ObjectIdDialog::onItemSelectionChanged);
+    connect(mUi->tableWidget, &QTableWidget::itemDoubleClicked, this, &ObjectIdDialog::onItemDoubleClicked);
+    connect(mUi->pushButton, &QPushButton::clicked, this, &ObjectIdDialog::onButtonClicked);
+}
+
+ObjectIdDialog::~ObjectIdDialog()
+{
+    Utils::saveGeometry(this);
+    delete mUi;
+}
+
+void ObjectIdDialog::setId(const int id)
+{
+    mId = id;
+
+    QLineEdit *lineEdit = mUi->lineEdit;
+    QTableWidget *tableWidget = mUi->tableWidget;
+    tableWidget->clearSelection();
+
+    QList<QTableWidgetItem *> items =
+            tableWidget->findItems(QString::number(mId), Qt::MatchExactly);
+
+    if (items.count() > 0) {
+        foreach (QTableWidgetItem *item, items) {
+            if (item->column() == 0) {
+                tableWidget->selectRow(item->row());
+                lineEdit->setText(QStringLiteral(""));
```

Comment:
Please use `QString()` to make an empty string. However, in this case, you should use `lineEdit->clear()`.